lib/omniauth/strategies/apple.rb

```diff
@@ -63,6 +63,20 @@ def callback_url
         options[:redirect_uri] || (full_host + callback_path)
       end
 
+      def callback_phase
+        if request.request_method.downcase.to_sym == :post
```

Comment:
Assuming we're solving this; https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/configuring_your_webpage_for_sign_in_with_apple#3331292